src/virtualenv/activation/batch/__init__.py

```diff
@@ -21,3 +21,9 @@ def instantiate_template(self, replacements, template, creator):
         # ensure the text has all newlines as \r\n - required by batch
         base = super(BatchActivator, self).instantiate_template(replacements, template, creator)
         return base.replace(os.linesep, "\n").replace("\n", os.linesep)
+
+    def replacements(self, creator, dest_folder):
+        base_replacements = super(BatchActivator, self).replacements(creator, dest_folder)
+        if not base_replacements["__VIRTUAL_PROMPT__"]:
+            base_replacements["__VIRTUAL_PROMPT__"] = "(%s) " % creator.env_name
```

Comment:
Shouldn't this change be in the template? Not the parameterization?

Reply:
I am not sure what do you mean, I will try to explain my reasoning:

In all other shells, there is a check in the shell script if the value replaced by virtual prompt is empty, and if so, it is set to the `env_name` in bat it is hard to do it, at least to my limited expertise in bat programing. To overcome this I have made the check here.

I do not think this should be in the `via_template.py` file if that what you mean, I see no need to complicate other shells because of bat limitations.

Reply:
A possible different change would be to change
https://github.com/pypa/virtualenv/blob/7649968fcc673336e510a18d4d1d20feef48fe4b/src/virtualenv/activation/via_template.py#L34
```python
 "__VIRTUAL_PROMPT__": "" if self.flag_prompt is None else self.flag_prompt,
```
to

````python
 "__VIRTUAL_PROMPT__": '(%s) ' % creator.env if self.flag_prompt is None else self.flag_prompt,
```
But I am not sure and will be unable to check how it effects all other shells, and the only ones I have access to are cmd.exe and (git) bash (both on windows).

Reply:
> 
> 
> Was thinking personally this should be a change within activate.bat 🤔

Correctly checking that a variable is empty and getting the folder name in bat is rather ugly, but I can try doing it, I'll need to learn how to. Seemed to me cleaner that way, similar to what have been in the previous version of vritualenv - the windows bat case was handled in python.

Reply:
Leave it as it is then, will investigate next week 👍

Reply:
Turns out fixing it it in the bat file, at least for the simple case was not so hard. Although, I can't say I understand why is this exact loop works and other I tried failed, so I'm not sure that it will work in edge cases, if there are.

I hope it's better now